src/ripple/app/misc/NetworkOPs.cpp

```diff
@@ -1399,25 +1399,43 @@ bool NetworkOPsImp::checkLastClosedLedger (
 }
 
 void NetworkOPsImp::switchLastClosedLedger (
-    Ledger::pointer newLedger, bool duringConsensus)
+    Ledger::pointer newLCL, bool duringConsensus)
 {
-    // set the newledger as our last closed ledger -- this is abnormal code
+    // set the newLCL as our last closed ledger -- this is abnormal code
 
     auto msg = duringConsensus ? "JUMPdc" : "JUMP";
-    m_journal.error << msg << " last closed ledger to " << newLedger->getHash ();
+    m_journal.error << msg << " last closed ledger to " << newLCL->getHash ();
 
     clearNeedNetworkLedger ();
-    newLedger->setClosed ();
-    auto openLedger = std::make_shared<Ledger> (false, std::ref (*newLedger));
-    m_ledgerMaster.switchLedgers (newLedger, openLedger);
+    newLCL->setClosed ();
+    auto newOL = std::make_shared<Ledger> (false, std::ref (*newLCL));
+    // Caller must own master lock
+    {
+        // Apply tx in old open ledger to new
+        // open ledger. Then apply local tx.
+        auto const oldOL =
+            m_ledgerMaster.getCurrentLedger();
+        // VFALCO What do we use for the hash?
+        CanonicalTXSet retries({});
```

Comment:
So `CanonicalTxSet` _doesn't ever use its argument._

I'll send out a change to get rid of the constructor, but feel free to use any hash you like here.